src/unlink.js

```diff
@@ -89,10 +89,14 @@ module.exports = function unlink(config, args) {
   unlinkDependencyIOS(project.ios, dependency, packageName);
 
   const allDependencies = getDependencyConfig(config, getProjectDependencies());
+  const otherDependencies = filter(allDependencies, ({name}) => name !== packageName);
```

Comment:
`allDependencies` may include the one being unlinked if it's still listed in the `package.json`